libsolidity/Parser.cpp

```diff
@@ -792,7 +792,15 @@ ASTPointer<Expression> Parser::parseLeftHandSideExpression(
 			std::tie(arguments, names) = parseFunctionCallArguments();
 			nodeFactory.markEndPosition();
 			expectToken(Token::RParen);
-			expression = nodeFactory.createNode<FunctionCall>(expression, arguments, names);
+			if (Identifier* identifier = dynamic_cast<Identifier*>(expression.get()))
+			{
+				ASTPointer<FunctionIdentifier> functionIdentifier = make_shared<FunctionIdentifier>(identifier->getLocation(), make_shared<ASTString>(identifier->getName()));
+				auto functionCall = nodeFactory.createNode<FunctionCall>(functionIdentifier, arguments, names);
+				functionIdentifier->setFunctionCall(functionCall);
```

Comment:
This is the only place in the parser where a node is modified right after it was created (and not only during the "resole names and types" or "checkTypeRequirements" stage). Can this be done differently?